ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -67,6 +67,10 @@
  * become eligible to be fetched.
  */
 class GetBlobOperation extends GetOperation<ReadableStreamChannel> {
+  // the options associated with the operation.
+  private final GetBlobOptions options;
+  // true if this operation represents a range request.
+  private final boolean hasRange;
```

Comment:
do you really need this variable? Not trying to nitpick but if `GetBlobOptions` evolves to have many options, you wouldn't want a `boolean` for every option right? If `GetBlobOptions` has a `hasRange()` function (or a `null` return on `getRange()` like what you have here), I don't see the need for this variable.

Just trying to declutter and cull class variables that aren't absolutely necessary.